subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,453 @@
+
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="manifest" dir="userguide/tutorial/manifest" title="Customization of MANIFEST.MF">
+    <sourcefile file="build.gradle" snippet="add-to-manifest"/>
+</sample>
+++++
+
+See api:org.gradle.api.java.archives.Manifest[] for the configuration options it provides.
+
+You can also create standalone instances of `Manifest`. One reason for doing so is to share manifest information between JARs. The following example demonstrates how to share common attributes between JARs:
+
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="manifest" dir="userguide/tutorial/manifest" title="Creating a manifest object.">
+    <sourcefile file="build.gradle" snippet="custom-manifest"/>
+</sample>
+++++
+
+Another option available to you is to merge manifests into a single `Manifest` object. Those source manifests can take the form of a text for or another `Manifest` object. In the following example, the source manifests are all text files except for `sharedManifest`, which is the `Manifest` object from the previous example:
+
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="manifest" dir="userguide/tutorial/manifest" title="Separate MANIFEST.MF for a particular archive">
+    <sourcefile file="build.gradle" snippet="merge"/>
+</sample>
+++++
+
+Manifests are merged in the order they are declared in the `from` statement. If the base manifest and the merged manifest both define values for the same key, the merged manifest wins by default. You can fully customize the merge behavior by adding `eachEntry` actions in which you have access to a api:org.gradle.api.java.archives.ManifestMergeDetails[] instance for each entry of the resulting manifest. Note that the merge is done lazily, either when generating the JAR or when `Manifest.writeTo()` or `Manifest.getEffectiveManifest()` are called.
+
+Speaking of `writeTo()`, you can use that to easily write a manifest to disk at any time, like so:
+
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="manifest" dir="userguide/tutorial/manifest" title="Saving a MANIFEST.MF to disk">
+    <sourcefile file="build.gradle" snippet="write"/>
+</sample>
+++++
+
+[[sec:generating_javadocs]]
+=== Generating API documentation
+
+The Java Plugin provides a `javadoc` task of type api:org.gradle.api.tasks.javadoc.Javadoc[], that will generate standard Javadocs for all your production code, i.e. whatever source is in the `main` source set. The task supports the core Javadoc and standard doclet options described in the http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#options[Javadoc reference documentation]. See api:org.gradle.external.javadoc.CoreJavadocOptions[] and api:org.gradle.external.javadoc.StandardJavadocDocletOptions[] for a complete list of those options.TODO:Not sure if the user can rely on `options` being of type `StandardJavadocDocletOptions` rather than `MinimalJavadocOptions`
```

Comment:
note there is a hanging TODO here

Reply:
That's because I don't have an answer to it yet. Was hoping one of the reviewers could provide one 😉

Reply:
Note that there is a hanging TODO in the Java testing chapter as well. The content was there originally, but I don't understand it. Unfortunately I haven't got an answer to that one yet.